### PR TITLE
add patch to fix catalog vuln

### PR DESCRIPTION
**Pull request by codebytere**
Add patch based on changes from 
https://chromium-review.googlesource.com/c/chromium/src/+/804397 
to fix 
https://bugs.chromium.org/p/project-zero/issues/detail?id=1450

/cc @alexeykuzmin @deepak1556

### CODE REVIEW

**Comment by alexeykuzmin** [comment]
All the other changed `manifest.json` files do not exist in the tree, right?

**Comment by codebytere** [comment]
nope, i removed the ones that failed with `file not found`

**Comment by alexeykuzmin** [comment]
[mash/quick_launch/manifest.json](https://chromium.googlesource.com/chromium/src/+/61.0.3163.100/mash/quick_launch/manifest.json) and [services/ui/manifest.json](https://chromium.googlesource.com/chromium/src/+/61.0.3163.100/services/ui/manifest.json) are there.
I'm not sure if we use them though.

**Comment by codebytere** [comment]
for some reason it's failing to apply on `mash/quick_launch/manifest.json `, i added `services/ui/manifest.json` just now

**Comment by alexeykuzmin** [comment]
Btw, can you please add the patch description to the patch file itself to make @alespergl happy? )

File: patches/076_patch_catalog_vuln.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+   InterfaceProviderSpec spec;
+-  spec.provides["app"].insert("filesystem::mojom::Directory");
++  spec.provides["directory"].insert("filesystem::mojom::Directory");
+   spec.provides["catalog:catalog"].insert("catalog::mojom::Catalog");
+   spec.provides["control"].insert("catalog::mojom::CatalogControl");
+   InterfaceProviderSpecMap specs;
+diff --git a/mash/quick_launch/manifest.json b/mash/quick_launch/manifest.json
+index 8dfbf4a..ac63cbe 100644
+--- a/mash/quick_launch/manifest.json
++++ b/mash/quick_launch/manifest.json
```
Comment (deepak1556) [inline patches/076_patch_catalog_vuln.patch]: We don't use mash services.

**Comment by deepak1556** [comment]
> mash/quick_launch/manifest.json and services/ui/manifest.json are there. I'm not sure if we use them though.

@alexeykuzmin content browser uses //services/ui https://cs.chromium.org/chromium/src/content/public/app/mojo/content_browser_manifest.json?dr&l=99 and mash service are used only for chromeOS at the moment.

**Comment by alexeykuzmin** [comment]
rel electron/electron#11916

**Pull request merged**